src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -102,26 +102,27 @@ private static void CreateCommandBuilders(CodeElement currentElement)
         {
             // Remove request executor
             RemoveUnusedParameters(currentClass);
-            // Replace Nav Properties with BuildXXXCommand methods
-            var navProperties = currentClass.GetChildElements().OfType<CodeProperty>().Where(e => e.IsOfKind(CodePropertyKind.RequestBuilder));
-            foreach (var navProp in navProperties)
-            {
-                var method = CreateBuildCommandMethod(navProp, currentClass);
-                currentClass.AddMethod(method);
-                currentClass.RemoveChildElement(navProp);
-            }
 
-            // Build command for indexers
-            var indexers = currentClass.GetChildElements().OfType<CodeIndexer>();
+            var children = currentClass.GetChildElements();
+            var indexers = children.OfType<CodeIndexer>();
```

Comment:
A class can only ever have a single indexer. replace by currentClass.Indexer and update accordingly.